addons/product/models/product_template.py

```diff
@@ -608,20 +611,33 @@ def action_open_documents(self):
                 'default_res_id': self.id,
                 'default_company_id': self.company_id.id,
             },
-            'domain': [('res_id', 'in', self.ids), ('res_model', '=', self._name)],
+            'domain': [
+                '|',
+                    '&', ('res_model', '=', 'product.template'), ('res_id', '=', self.id),
+                    '&',
+                        ('res_model', '=', 'product.product'),
+                        ('res_id', 'in', self.product_variant_ids.ids),
+            ],
```

Comment:
This domain should be replaced when `action_open_documents` is called from the `product.product`, the domain should only include the template document or the variant ones imho, not all the documents of the template variants :thinking: 
![image](https://github.com/odoo/odoo/assets/19364378/5cd92db0-b124-41f8-94aa-f396b801501b)

Reply:
That needs to be discussed with @dalaOdoo. From what I recall, the idea is that the filter `search_default_context_variant` is applied first to restrict to the current variant, but if you remove the filter you'd like a global view on all products from the related template.

Reply:
Yeah, I don't strictly mind, but I wanted to be sure that's really the expected behavior :ok_hand:

Reply:
If you remove the "Documents of this variant" filter when coming from a variant, what do you expect to see ?
Documents of the template and of the variant you are coming from or the documents of all variants of the template ?

Reply:
Theoretically, only the documents that matters for this variant so the ones of the template (no of other variants) but practically I feel like displaying everything may help the user : to duplicate, to find a document wrongly configured, ...

Reply:
Mhm, ok then, let's keep it like this.